client/src/main/java/org/eclipse/hono/client/impl/CredentialsClientImpl.java

```diff
@@ -63,9 +59,16 @@ protected String createMessageId() {
     }
 
     @Override
-    protected CredentialsResult getResult(final int status, final JsonObject payload) {
-
-        return CredentialsResult.from(status, payload);
+    protected CredentialsResult<CredentialsObject> getResult(final int status, final String payload) {
+        try {
+            if (status == HTTP_OK) {
+                ObjectMapper om = new ObjectMapper();
```

Comment:
Very good point, just checked: it is definitely thread-safe, so I will declare it statically in the using class.
Good explanation about ObjectMapper and how to use it best found under
https://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field